src/main/java/teammates/ui/newcontroller/SearchStudentsAndFeedbackSessionDataAction.java

```diff
@@ -0,0 +1,154 @@
+package teammates.ui.newcontroller;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import teammates.common.datatransfer.FeedbackResponseCommentSearchResultBundle;
+import teammates.common.datatransfer.StudentSearchResultBundle;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.Const;
+
+/**
+ * Action: Showing the InstructorSearchPage for an instructor.
```

Comment:
The API should not have any knowledge on the "page" or even know what a "page" is.